x-pack/plugins/task_manager/server/lib/retryable_bulk_update.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectError } from '@kbn/core-saved-objects-common';
+import { ConcreteTaskInstance } from '../task';
+import { TaskStore, BulkUpdateResult, BulkGetResult } from '../task_store';
+import { isErr, isOk, asErr } from './result_type';
+import { BulkUpdateTaskResult } from '../task_scheduling';
+
+export const MAX_RETRIES = 2;
+
+export interface RetryableBulkUpdateOpts {
+  taskIds: string[];
+  getTasks: (taskIds: string[]) => Promise<BulkGetResult>;
+  filter: (task: ConcreteTaskInstance) => boolean;
+  map: (task: ConcreteTaskInstance) => ConcreteTaskInstance;
+  store: TaskStore;
+}
+
+export async function retryableBulkUpdate({
+  taskIds,
+  getTasks,
+  filter,
+  map,
+  store,
+}: RetryableBulkUpdateOpts): Promise<BulkUpdateTaskResult> {
```

Comment:
This is the re-usable function to retry bulk updates by doing the following steps:
1. Fetch data
2. Filter out successful fetched documents
3. Call the filter function passed-in (ex: to filter out enabled tasks)
4. Call the map function passed-in (ex: to set enable to true)
5. Call the task store's bulk update API
6. Repeat steps 1 to 5 on the documents that encountered a 409 conflict